tests/test_dispatcher.py

```diff
@@ -730,6 +730,51 @@ def test_sensible_worker_thread_names(self, dp2):
         for thread_name in thread_names:
             assert thread_name.startswith(f"Bot:{dp2.bot.id}:worker:")
 
+    @pytest.mark.parametrize(
+        'message',
+        [
+            Message(message_id=1, chat=Chat(id=2, type=None), migrate_from_chat_id=1, date=None),
+            Message(message_id=1, chat=Chat(id=1, type=None), migrate_to_chat_id=2, date=None),
+            Message(message_id=1, chat=Chat(id=1, type=None), date=None),
+            None,
+        ],
+    )
+    @pytest.mark.parametrize('old_chat_id', [None, 1, "1"])
+    @pytest.mark.parametrize('new_chat_id', [None, 2, "1"])
+    def test_migrate_chat_data(self, dp, message: 'Message', old_chat_id: int, new_chat_id: int):
+        def call(match: str):
+            with pytest.raises(ValueError, match=match):
+                dp.migrate_chat_data(
+                    message=message, old_chat_id=old_chat_id, new_chat_id=new_chat_id
+                )
+
+        if message and (old_chat_id or new_chat_id):
+            call(r"^Message and chat_id pair are mutually exclusive$")
+            return
+
+        if not any((message, old_chat_id, new_chat_id)):
+            call(r"^chat_id pair or message must be passed$")
+            return
+
+        if message:
+            if message.migrate_from_chat_id is None and message.migrate_to_chat_id is None:
+                call(r"^Invalid message istance, can't detect old/new chat_id$")
+                return
+            _old_chat_id = message.migrate_from_chat_id or message.chat.id
+            _new_chat_id = message.migrate_to_chat_id or message.chat.id
+
+        elif not (isinstance(old_chat_id, int) and isinstance(new_chat_id, int)):
+            call(r"^old_chat_id and new_chat_id must be integers$")
+            return
+        else:
+            _old_chat_id = old_chat_id
+            _new_chat_id = new_chat_id
```

Comment:
what's this good for? IISC we can just skip this

Reply:
> what's this good for? IISC we can just skip this

old_chat_id and new_chat_id can be None,  for the final assert I need of valid ids, i am mistaking something? If no in the row [775 ](https://github.com/DonalDuck004/python-telegram-bot/blob/migrate_chat_data/tests/test_dispatcher.py#L775) I made an error: /

Reply:
Ah, now I understand - you're right! maybe rename to `effective_*_id`? may be a bit more speaking name :)

Reply:
Done